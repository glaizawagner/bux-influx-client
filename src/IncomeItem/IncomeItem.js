/* eslint-disable no-unused-expressions */
import React from 'react';
import BuxInfluxContext from '../BuxInfluxContext';
// import PropTypes from 'prop-types';
import config from '../config'

function deleteIncomeRequest(iid, cb) {
    fetch(config.API_ENDPOINT +`/{iid}`, {
        method: 'DELETE',
        headers: {
            'content-type': 'application/json',
            // 'authorization': `bearer ${config.API_KEY}`
        }
    })
        .then(res => {
            if (!res.ok) {
            return res.json().then(error => Promise.reject(error))
            }
            // return res.json()
        })
        .then(() => {
            cb(iid)
        })
        .catch(error => {
            console.error(error)
        })
}

export default function IncomeItem(props) {
    return(
        <BuxInfluxContext.Consumer>
            { (context) => (
                <li className ='IncomeItem'>
                    <div className = 'IncomeItem__row'>
                        <span> 
                            {props.date_created} 
                            {props.description} 
                            {props.value} 
                            <button 
                                className='IncomeItem__btn'
                                onClick={() => {
                                    deleteIncomeRequest(
                                        props.iid,
                                        context.deleteIncome
                                    )
                                }}
                            >
                            Delete
                            </button>
                        </span>
                    </div>
                </li>
            )}
        </BuxInfluxContext.Consumer>
    )
}

