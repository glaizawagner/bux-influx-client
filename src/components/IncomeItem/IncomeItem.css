.IncomeItem__row {
    padding: 0 3rem;
    margin: 0 2rem;
    display: flex;
    justify-content: space-between;
}