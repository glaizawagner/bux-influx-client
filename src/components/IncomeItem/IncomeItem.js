/* eslint-disable no-unused-expressions */
import React from 'react';
import BuxinfluxContext from '../../contexts/BuxinfluxContext';
import BuxinfluxApiService from '../../services/buxinflux-api-service'
// import PropTypes from 'prop-types';
// import config from '../../config'

function deleteIncomeRequest(iid,cb) {
    BuxinfluxApiService.deleteIncome(iid)
        .then(() => {
            cb(iid)
        })
        .catch((e) => console.error(e));
}

export default function IncomeItem(props) {

    return(
        <BuxinfluxContext.Consumer>
            { (context) => (
                <li className ='IncomeItem'>
                    <div className = 'IncomeItem__row'>
                        <span> 
                            {props.date_created} 
                            {props.description} 
                            {props.value} 
        
                            <button 
                                className='IncomeItem__description'
                                onClick={() => {
                                    deleteIncomeRequest(
                                        props.iid,
                                        context.deleteIncome
                                    )
                                }}
                            >
                            Delete
                            </button>
                        </span>
                    </div>
                </li>
            )}
        </BuxinfluxContext.Consumer>
    )
}

