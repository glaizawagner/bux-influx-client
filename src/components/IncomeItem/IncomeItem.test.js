import React from 'react';
import { shallow } from 'enzyme';
import IncomeItem from './IncomeItem';
import toJson from 'enzyme-to-json'

describe('Income Item  component', () => {
    const props = {
        iid: 1,
        date_created: '2019-11-12T16:28:32.615Z',
        type: 'inc',
        description: 'Salary',
        value: '3500.00',
        user_id: 1
    }
    
    it('renders Income by default', () => {
        const wrapper = shallow(<IncomeItem {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders Income given props', () => {
        const wrapper = shallow(<IncomeItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});