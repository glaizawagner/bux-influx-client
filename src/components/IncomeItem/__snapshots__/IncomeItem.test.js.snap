// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Income Item  component renders Income by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Income Item  component renders Income given props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
